File: DesignPatternsExercises/ruby/abstract_factory_design_pattern/exercise/abstract_factory_design_pattern.rb
require_relative 'abstract_factory'

phone_number_factory = AbstractFactory.factory(:Phone)
phone_number_uk = phone_number_factory.instance(:UK, '+44 7836 191 191')
phone_number_us = phone_number_factory.instance(:US, '(919) 515-2011')
puts phone_number_uk.phone_number
puts phone_number_us.phone_number
puts ''

post_code_factory = AbstractFactory.factory(:Post)
post_code_uk = post_code_factory.instance(:UK, 'SW15 5PU')
post_code_us = post_code_factory.instance(:US, '27606')
puts post_code_uk.post_code
puts post_code_us.post_code

File: DesignPatternsExercises/java/AbstractFactoryDesignPattern/Exercise/AbstractFactoryDesignPattern.java
public class AbstractFactoryDesignPattern {
    public static void main(String[] args) {
        AbstractFactory phoneNumberFactory = PostPhoneFactory.getFactory(Type.PHONE);
        PhoneNumber phoneNumberUK = phoneNumberFactory.getPhoneNumberInstance(Country.UK, "+44 7836 191 191");
        PhoneNumber phoneNumberUS = phoneNumberFactory.getPhoneNumberInstance(Country.US, "(919) 515-2011");
        System.out.println(phoneNumberUK.getPhoneNumber());
        System.out.println(phoneNumberUS.getPhoneNumber());
        System.out.println();

        AbstractFactory postCodeFactory = PostPhoneFactory.getFactory(Type.POST);
        PostCode postCodeUK = postCodeFactory.getPostCodeInstance(Country.UK, "SW15 5PU");
        PostCode postCodeUS = postCodeFactory.getPostCodeInstance(Country.US, "27606");
        System.out.println(postCodeUK.getPostalCode());
        System.out.println(postCodeUS.getPostalCode());
    }
}
